nautilus-extension/nautilus-gsconnect.py

```diff
@@ -2,85 +2,75 @@
 #
-# Nemo seems to reliably identify itself as 'nemo' in argv[0], so we
+# Nemo seems to reliably identify itself as "nemo" in argv[0], so we
 # can test for that. Nautilus detection is less reliable, so don't try.
 # See https://github.com/linuxmint/nemo-extensions/issues/330
-if "nemo" in sys.argv[0].lower():
+if "nemo" in argv[0].lower():
     # Host runtime is nemo-python
-    gi.require_version('Nemo', '3.0')
+    gi.require_version("Nemo", "3.0")
     from gi.repository import Nemo as FileManager
 else:
     # Otherwise, just assume it's nautilus-python
     from gi.repository import Nautilus as FileManager
 
 
-SERVICE_NAME = 'org.gnome.Shell.Extensions.GSConnect'
-SERVICE_PATH = '/org/gnome/Shell/Extensions/GSConnect'
+SERVICE_NAME = "org.gnome.Shell.Extensions.GSConnect"
+SERVICE_PATH = "/org/gnome/Shell/Extensions/GSConnect"
 
 # Init gettext translations
-LOCALE_DIR = os.path.join(GLib.get_user_data_dir(),
-                          'gnome-shell', 'extensions',
-                          'gsconnect@andyholmes.github.io', 'locale')
+LOCALE_DIR = join(
+    get_user_data_dir(), "gnome-shell", "extensions",
+    "gsconnect@andyholmes.github.io", "locale")
 
-if not os.path.exists(LOCALE_DIR):
+if not exists(LOCALE_DIR):
     LOCALE_DIR = None
 
 try:
-    i18n = gettext.translation(SERVICE_NAME,
-                               localedir=LOCALE_DIR)
+    i18n = translation(SERVICE_NAME, localedir=LOCALE_DIR)
     _ = i18n.gettext
-
-except (IOError, OSError) as e:
-    print('GSConnect: {0}'.format(e.strerror))
-    i18n = gettext.translation(SERVICE_NAME,
-                               localedir=LOCALE_DIR,
-                               fallback=True)
+except (IOError, OSError) as error:
+    print(f"GSConnect: {error.strerror}")
+    i18n = translation(SERVICE_NAME, localedir=LOCALE_DIR, fallback=True)
     _ = i18n.gettext
 
 
-class GSConnectShareExtension(GObject.Object, FileManager.MenuProvider):
+class GSConnectShareExtension(Object, FileManager.MenuProvider):
     """A context menu for sending files via GSConnect."""
 
     def __init__(self):
         """Initialize the DBus ObjectManager"""
-
-        GObject.Object.__init__(self)
-
-        self.devices = {}
-
-        Gio.DBusProxy.new_for_bus(Gio.BusType.SESSION,
-                                  Gio.DBusProxyFlags.DO_NOT_AUTO_START,
-                                  None,
-                                  SERVICE_NAME,
-                                  SERVICE_PATH,
-                                  'org.freedesktop.DBus.ObjectManager',
-                                  None,
-                                  self._init_async,
-                                  None)
+        Object.__init__(self)
+        self.devices = dict()
+        DBusProxy.new_for_bus(
+            BusType.SESSION, DBusProxyFlags.DO_NOT_AUTO_START, None,
+            SERVICE_NAME, SERVICE_PATH, "org.freedesktop.DBus.ObjectManager",
+            None, self._init_async, None)
```

Comment:
No. Hard no from me. For the wrapping alone.

This packed style is not "more readable", in fact it's far easier to lose track of which argument is which. I'd fully support wrapping the args under the call (that aligned-crazy-far-to-the-right style drives me nuts, TBH), but the one-arg-per-line formatting is extremely valuable given how many there are. It also makes diffs that change any parameters much cleaner.

Reply:
(An example of the cleaner-diffs issue: The third arg to `DBusProxy.new_for_bus()` is an optional `GDBusInterfaceInfo`. Let's say we need to start passing one of those in for some reason.

With the previous code, that diff looks like this:
```diff
@@ -4,7 +4,7 @@
 
         Gio.DBusProxy.new_for_bus(Gio.BusType.SESSION,
                                   Gio.DBusProxyFlags.DO_NOT_AUTO_START,
-                                  None,
+                                  dbus_interface_info,
                                   SERVICE_NAME,
                                   SERVICE_PATH,
                                   'org.freedesktop.DBus.ObjectManager',
```

The packed version? This happens:
```diff
@@ -16,10 +16,10 @@
         Object.__init__(self)
         self.devices = dict()
         DBusProxy.new_for_bus(
-            BusType.SESSION, DBusProxyFlags.DO_NOT_AUTO_START, None,
-            SERVICE_NAME, SERVICE_PATH, "org.freedesktop.DBus.ObjectManager",
-            None, self._init_async, None)
+            BusType.SESSION, DBusProxyFlags.DO_NOT_AUTO_START,
+            dbus_interface_info, SERVICE_NAME, SERVICE_PATH,
+            "org.freedesktop.DBus.ObjectManager", None,
+            self._init_async, None)
```